src/code/z_bgcheck.c

```diff
@@ -1,432 +1,4713 @@
+s32 BgCheck_PosInStaticBoundingBox(CollisionContext* colCtx, Vec3f* pos);
+
+void SSNode_SetValue(SSNode* node, s16* polyIndex, u16 next) {
     node->polyId = *polyIndex;
     node->next = next;
 }
 
-void BgCheck_PolygonLinkedListResetHead(u16* head) {
-    *head = 0xFFFF;
+void SSList_SetNull(SSList* ssList) {
+    ssList->head = SS_NULL;
 }
 
-void BgCheck_ScenePolygonListsNodeInsert(SSNodeList* list, u16* head, s16* polyIndex) {
+void SSNodeList_SetSSListHead(SSNodeList* list, SSList* ssList, s16* polyIndex) {
     u16 index;
 
-    index = BgCheck_ScenePolygonListsReserveNode(list);
-    BgCheck_PolygonLinkedListNodeInit(&list->tbl[index], polyIndex, *head);
-    *head = index;
+    index = SSNodeList_GetNextNodeIdx(list);
+    SSNode_SetValue(&list->tbl[index], polyIndex, ssList->head);
+    ssList->head = index;
 }
 
-void BgCheck_PolygonLinkedListNodeInsert(DynaSSNodeList* list, u16* head, s16* polyIndex) {
+void DynaSSNodeList_SetSSListHead(DynaSSNodeList* list, SSList* ssList, s16* polyIndex) {
     u16 index;
 
-    index = BgCheck_AllocPolygonLinkedListNode(list);
-    BgCheck_PolygonLinkedListNodeInit(&list->tbl[index], polyIndex, *head);
-    *head = index;
+    index = DynaSSNodeList_GetNextNodeIdx(list);
+    SSNode_SetValue(&list->tbl[index], polyIndex, ssList->head);
+    ssList->head = index;
 }
 
-void BgCheck_PolygonLinkedListInit(GlobalContext* globalCtx, DynaSSNodeList* list) {
+void DynaSSNodeList_Init(GlobalContext* globalCtx, DynaSSNodeList* list) {
     list->tbl = NULL;
     list->count = 0;
 }
 
-void BgCheck_PolygonLinkedListAlloc(GlobalContext* globalCtx, DynaSSNodeList* list, u32 numNodes) {
-    list->tbl = (SSNode*)THA_AllocEndAlign(&globalCtx->state.heap, numNodes << 2, 0xfffffffe);
-    list->max = numNodes;
+void DynaSSNodeList_Alloc(GlobalContext* globalCtx, DynaSSNodeList* list, u32 numNodes) {
+    list->tbl = (SSNode*)THA_AllocEndAlign(&globalCtx->state.heap, numNodes * sizeof(SSNode), -2);
+    list->maxNodes = numNodes;
     list->count = 0;
 }
 
-void BgCheck_PolygonLinkedListReset(DynaSSNodeList* list) {
+void DynaSSNodeList_ResetCount(DynaSSNodeList* list) {
     list->count = 0;
 }
 
-u16 BgCheck_AllocPolygonLinkedListNode(DynaSSNodeList* list) {
-    u16 index;
+u16 DynaSSNodeList_GetNextNodeIdx(DynaSSNodeList* list) {
+    u16 index = list->count++;
 
-    index = list->count++;
-    if (list->max <= index) {
-        return 0xffff;
+    if (list->maxNodes <= index) {
+        return SS_NULL;
     }
 
     return index;
 }
 
-void BgCheck_CreateVec3fFromVertex(Vec3s* vertex, Vec3f* vector) {
+void BgCheck_Vec3sToVec3f(Vec3s* vertex, Vec3f* vector) {
     vector->x = vertex->x;
     vector->y = vertex->y;
     vector->z = vertex->z;
 }
 
-void BgCheck_CreateVertexFromVec3f(Vec3s* vertex, Vec3f* vector) {
+void BgCheck_Vec3fToVec3s(Vec3s* vertex, Vec3f* vector) {
     vertex->x = vector->x;
     vertex->y = vector->y;
     vertex->z = vector->z;
 }
```

Comment:
```suggestion
void BgCheck_Vec3fToVec3s(Vec3s* dest, Vec3f* src) {
    dest->x = src->x;
    dest->y = src->y;
    dest->z = src->z;
}
```
Same here. Although it might be fine as is. I think this is more consistent with how it's done in other parts of the repo